### ISSUE TITLE
Getting "should not reach" error

### ISSUE THREAD

**Issue by emilyastranova** [blackd/Inventory-Profiles#29]
Getting java.lang.IllegalStateException: should not reach here when trying to sort my inventory

![image](https://user-images.githubusercontent.com/16725119/127096055-1a464fb3-15a8-48ec-8263-78b425d62a15.png)

**Comment by blackd** [comment]
Hi,
What are:
- the game version
- the mod version
- the mod loader (fabric or forge)

Also please send the full logs i.e. the 2 newest files from: 
- on windows press Wind+R and paste "%appdata%\.minecraft\logs\ "
- on mac open Finder then paste: ~/Library/Application Support/minecraft in the Search box
- on linux ~\,minecraft\logs

**Comment by emilyastranova** [comment]
[2021-07-25-3.log.gz](https://github.com/blackd/Inventory-Profiles/files/6882869/2021-07-25-3.log.gz)
[2021-07-25-2.log.gz](https://github.com/blackd/Inventory-Profiles/files/6882870/2021-07-25-2.log.gz)

Game version: 1.16.5
Mod version: 0.8.4
Mod loader: Forge

**Comment by blackd** [comment]
I guess the logs are old? Because they show you don't have the mod installed.

**Comment by emilyastranova** [comment]
[2021-07-26-1.log.gz](https://github.com/blackd/Inventory-Profiles/files/6882908/2021-07-26-1.log.gz)

Sorry I found the relevant one with the error in it. 🤦‍♂️

**Comment by blackd** [comment]
Looks like mod to mod incompatibility. If you haven't changed any other mod since the first logs you've sent I'll try to recreate the situation.
If you have I guess send me a list :)
Also if you want you can join our discord server https://discord.gg/23YCxmveUM for faster communication.

**Comment by emilyastranova** [comment]
Sorry for the slow responses, having fun with the pack right now. Here is the list of mods:
```
alexsmobs-1.8.1.jar
appleskin-forge-mc1.16.x-2.1.0.jar
Aquaculture-1.16.5-2.1.16.jar
archers_paradox-1.16.4-1.2.0.jar
areas_1.16.5-2.2.jar
BackTools-1.16.5-10.1.0.jar
BetterCaves-Forge-1.16.4-1.1.1.jar
betterendforge-1.16.4-1.5.1.jar
BetterMineshafts-Forge-1.16.4-2.0.3.jar
Bountiful-1.16.4-3.3.1.jar
byg-1.1.14.jar
carryon-1.16.5-1.15.5.15.jar
cfm-7.0.0pre22-1.16.3.jar
citadel-1.6.2.jar
cloth-config-4.11.26-forge.jar
cofh_core-1.16.4-1.2.1.jar
-collective-1.16.5-2.25.jar
comforts-forge-1.16.4-4.0.1.0.jar
corpse-1.16.5-1.0.2.jar
DungeonCrawl-1.16.3-2.2.4.jar
DynamicSurroundings-1.16.4-4.0.4.1.jar
EasyMagic-v1.0.2-1.16.5.jar
EndRemastered-R36.jar
enhancedcelestials-1.0.2-1.16.4.jar
FastLeafDecay-v25.jar
ferritecore-2.0.5-forge.jar
firstjoinmessage_1.16.5-1.2.jar
guardvillagers-1.16.5.1.2.2.jar
Hwyla-forge-1.10.11-B78_1.16.2.jar
iceandfire-2.1.6-1.16.4.jar
iChunUtil-1.16.3-10.0.0.jar
InventoryHud_[1.16.2-1.16.5].forge-3.2.2.jar
InventoryProfilesNext-forge-1.16-0.8.4.jar
jei-1.16.5-7.6.1.75.jar
kotlinforforge-1.10.0-obf.jar
mcw-bridges-1.0.6-mc1.16.5.jar
mowziesmobs-1.5.15.jar
mysticalworld-1.16.4-0.3.1.9.jar
NetherPortalFix_1.16.3-7.2.1.jar
netherportalspread_1.16.5-5.1.jar
OptiFine_1.16.5_HD_U_G8_MOD.jar
OreExcavation-1.8.157.jar
Paraglider-1.16.2-1.3.1.0.jar
Patchouli-1.16.4-50.jar
ProgressiveBosses-2.2.0-mc1.16.x.jar
projectvibrantjourneys-1.16.5-3.0.0.jar
quickteleports-1.16.2-1.1.2.jar
randomvillagenames_1.16.5-1.1.jar
RealisticTorches-1.16.5-3.1.2.jar
repurposed_structures-1.16.5-2.5.1.jar
SubWild-1.16.5-1.2.0.jar
towers_of_the_wild-1.16.4-2.0.1.jar
TrashSlot_1.16.3-12.2.1.jar
TravelersBackpack-1.16.5-5.4.0.jar
twoplayersonehorse-1.16.4-2.0.2.jar
valhelsia_structures-16.0.5.jar
villagernames_1.16.5-2.9.jar
villagespawnpoint_1.16.5-1.5.jar
walljump-forge-1.16.4-1.3.7.jar
Waystones_1.16.5-7.4.0.jar
Xaeros_Minimap_21.14.1_Forge_1.16.5.jar
YungsApi-1.16.4-Forge-4.jar
```

**Comment by emilyastranova** [comment]
I'm going to go out on a limb and assume its either the backpack mod or trash slot mod

**Comment by emilyastranova** [comment]
Should also be mentioned it as after lots of gameplay, and it goes away after a server restart I believe

**Comment by blackd** [comment]
You can try one of these
- set config->Gui Settings->Add inventory buttons to all in game screens to false (this will also disable the sorting of the backpack)
- Enable config->Mod Settings->Advanced Options->Add interval between clicks. And probably adjust the interval below

**Comment by marie-bnl** [comment]
I have this error, and I have only this mod and Optifine

**Comment by marie-bnl** [comment]
It seems to do that only when there are certain items in the chest at certain positions

![image](https://user-images.githubusercontent.com/39555268/127385504-b6f46455-c14a-4d98-a501-bf3d58fd6ace.png)

Often when I edit some things manually like removing an item, sorting, putting the item again and sorting again it works

**Comment by blackd** [comment]
Whatever I've tried I can't reproduce this.
I'll include more debug code in the next release to geгter some information whenever this happens so you can eventually post it here.

**Comment by marie-bnl** [comment]
Comment here when the next release is available and I'll post the logs here ^^

**Comment by marie-bnl** [comment]
Also it seems to do that also sometimes when the chest is full (but not only)

**Comment by blackd** [comment]
@tyleraharrison, @victorbnl  Have you notices if non stackable items are always involved, usually enchanted non stackable items? 

Also do you for example have stacked non stackable items :) because these cause failures too.

**Comment by marie-bnl** [comment]
>  Have you notices if non stackable items are always involved, usually enchanted non stackable items?

From what I remember it does that also with stackable items but I'm not sure at all, but you're right it's mainly with non-stackable ones.

> Also do you for example have stacked non stackable items :) because these cause failures too.

I play in survival with no mods or anything so I guess it wouldn't be possible, so no

It's very difficult to test as it doesn't only depend on the items themselves but also their position, but I'll try to make some tests and tell you if I find something relevant

Also it could be certain items that cause this, because once it made this error in my inventory and I dropped items one by one to see at which one it would stop making the error and it worked (it stopped at an enchanted book).

**Comment by marie-bnl** [comment]
Also here are the details on my config 

Game version : 1.16.5
Mod version : 0.8.5
Mod loader : Forge

**Comment by blackd** [comment]
success!!
if you have 1 Curse of Vanishing and 3 Bane of Arthropods IV books depending on the positions it's reproducible.

**Comment by marie-bnl** [comment]
If you take one or all of the items, sort and put them again in the position, does it happen again?

**Comment by blackd** [comment]
yes

**Comment by marie-bnl** [comment]
Okay good! Good luck for the fix

**Comment by emilyastranova** [comment]
I had enchanted items as well, this checks out

**Comment by marie-bnl** [comment]
I have a chest with which I have the error and I was able to do some tests

I can't do linear and row sortings, but I can do columns sorting.  
If I do column sorting, then I can do linear and row sortings.  
But if I do one, it goes again at the beginning of this sentence

**Comment by blackd** [comment]
Hi,

Do you have access to https://github.com/blackd/Inventory-Profiles/actions/runs/1084873489 and to the "dev-build" inside?

If you do please download it and test it if you still have chests that caused the error.
If you don't I'll upload it somewhere else.

**Comment by marie-bnl** [comment]
Yes I do, I'll test :+1:

**Comment by marie-bnl** [comment]
I still have the stable version and weirdly I can't reproduce it. Maybe just closing/reopening the game just made the issue go away for this chest? If I find this bug again I'll test

**Comment by marie-bnl** [comment]
Ok so I had the error again, I didn't touch the chest anymore and restarted the game to make sure it was not the cause in case it would work and tried with the dev build, and I can confirm this bug seems to be resolved!

**Comment by blackd** [comment]
Fixed in 2d11437
will be released in 0.8.5
